packages/ckeditor5-widget/tests/utils.js

```diff
@@ -502,6 +503,58 @@ describe( 'widget utils', () => {
 		} );
 	} );
 
+	describe( 'checkSelectionOnObject()', () => {
+		let model;
+
+		beforeEach( () => {
+			model = new Model();
+
+			model.document.createRoot();
+
+			model.schema.register( 'image', {
+				allowIn: '$root',
+				isObject: true,
+				isBlock: true
+			} );
+
+			model.schema.register( 'paragraph', {
+				inheritAllFrom: '$block'
+			} );
+
+			model.schema.register( 'element', {
+				allowIn: '$root',
+				isSelectable: true
+			} );
+		} );
+
+		it( 'should return false if no element is selected', () => {
+			setData( model, '<paragraph>[]</paragraph>' );
+
+			const selection = model.document.selection;
+			const isSelectionOnObject = checkSelectionOnObject( selection, model.schema );
+
+			expect( isSelectionOnObject ).to.be.false;
+		} );
+
+		it( 'should return false if the selection is not on the object', () => {
+			setData( model, '[<element></element>]' );
+
+			const selection = model.document.selection;
+			const isSelectionOnObject = checkSelectionOnObject( selection, model.schema );
+
+			expect( isSelectionOnObject ).to.be.false;
+		} );
+
+		it( 'should return true if the selection is on the object', () => {
+			setData( model, '<paragraph></paragraph>[<image></image>]' );
+
+			const selection = model.document.selection;
+			const isSelectionOnObject = checkSelectionOnObject( selection, model.schema );
+
+			expect( isSelectionOnObject ).to.be.true;
+		} );
+	} );
```

Comment:
I think we're missing two cases here

*   `<paragraph>fo[o</paragraph><image></image><paragraph>ba]r</paragraph>` 
*   `<image>[foo]</image>` (this will require allowing `$text` in `image`)